snap/quota/quota.go

```diff
@@ -153,6 +221,267 @@ func (grp *Group) SliceFileName() string {
 	return buf.String()
 }
 
+// groupQuotaAllocations contains information about current quotas of a group
+// and is used by getgroupQuotaAllocations to contain this information.
+// There are two types of values for each quota - the quota limit set by this group,
+// and the quota reserved by children of this group. Examples:
+// Group that has a non-memory quota, but has a child group that has a memory quota of 512mb:
+// memoryLimit = 0
+// memoryReserved = 512 mb
+// Group that has a memory quota of 512mb, but has only children groups with non-memory quota:
+// memoryLimit = 512 mb
+// memoryReserved = 0
+// Group that has a memory quota of 512mb, and has a child group that has a memory quota of 256mb:
+// memoryLimit = 512 mb
+// memoryReserved = 256 mb
+// If the limit value is non-zero, then the reserved value can never be greater than the limit, however
+// if the limit is zero, then the reserved value must be below the nearest non-zero limit as you traverse
+// up the tree.
+type GroupQuotaAllocations struct {
```

Comment:
this is not used outside of the package right? it should be not exported and then tests can do

```
type GroupQuotaAllocations = groupQuotaAllocations
```

in export_test.go

Reply:
Ah, right, thats cool, didn't realize that was an option!